tests/nativeapp/test_commands.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from unittest import mock
+from subprocess import CalledProcessError
+
+from tests.testing_utils.fixtures import *
+
+PROJECT_NAME = "demo_na_project"
+
+
+@mock.patch("pathlib.Path.is_file", return_value=True)
+def test_init_no_template_w_existing_yml(mock_path_is_file, runner):
+    result = runner.invoke(["app", "init", PROJECT_NAME])
+
+    assert (
+        "Cannot initialize a new project within an existing Native Application project!"
+        in result.output
+    )
+    assert result.exit_code != 0
+
+
+@mock.patch("pathlib.Path.exists", return_value=True)
+def test_init_no_template_w_existing_directory(mock_path_exists, runner):
+    result = runner.invoke(["app", "init", PROJECT_NAME])
+
+    assert (
+        f"This directory already contains a sub-directory called {PROJECT_NAME}. Please try a different name."
+        in result.output
+    )
```

Comment:
In tests we usually add `--format JSON`, it is much easier to assert json. Example in integration tests https://github.com/Snowflake-Labs/snowcli/blob/main/tests_integration/conftest.py#L55-L70 . Maybe add new `invoke_with_json_output` that add format flag and loads that json.